tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5095,7 +5095,7 @@ class HLSLExternalSource : public ExternalSemaSource {
               llvm::ArrayRef<TemplateArgument>(TST->getArgs(),
                                                TST->getNumArgs()));
         }
-        if (const RecordType* recordType = argType->getAsStructureType()) {
+        if (const RecordType* recordType = argType->getAsStructureOrClassType()) {
```

Comment:
Does this need to be only struct or class types, or could it just be `getAsRecordType()`?